modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/SnapshotMetadataVerificationTask.java

```diff
@@ -88,25 +96,74 @@ public MetadataVerificationJob(SnapshotMetadataVerificationTaskArg arg) {
         }
 
         /** {@inheritDoc} */
-        @Override public List<SnapshotMetadata> execute() throws IgniteException {
+        @Override public List<SnapshotMetadata> execute() {
             IgniteSnapshotManager snpMgr = ignite.context().cache().context().snapshotMgr();
 
-            List<SnapshotMetadata> snpMeta = snpMgr.readSnapshotMetadatas(arg.snapshotName(), arg.snapshotPath());
+            List<SnapshotMetadata> metas = snpMgr.readSnapshotMetadatas(arg.snapshotName(), arg.snapshotPath());
+
+            for (SnapshotMetadata meta : metas)
+                checkMeta(meta, ignite.context().config().getEncryptionSpi().masterKeyDigest());
 
             if (arg.incrementIndex() > 0) {
-                List<SnapshotMetadata> metas = snpMeta.stream()
+                List<SnapshotMetadata> locNodeMetas = metas.stream()
                     .filter(m -> m.consistentId().equals(ignite.localNode().consistentId()))
                     .collect(Collectors.toList());
 
-                if (metas.size() != 1) {
-                    throw new IgniteException("Failed to find snapshot metafile [metas=" + metas +
-                        ", snpName=" + arg.snapshotName() + ", snpPath=" + arg.snapshotPath() + ']');
+                if (locNodeMetas.size() != 1) {
+                    throw new IgniteException("Failed to find single snapshot metafile for local node [locNodeId="
```

Comment:
Let's write here that incremental snapshots requires single meta file per node, because incremental snapshots don't support restoring on different topology.